File: Arrays & Hashing/1. Contains Duplicate/solution.exs
defmodule Solution do
  @spec contains_duplicate(nums :: [integer]) :: boolean
  def contains_duplicate(nums) do
    recursive(nums, %{})
  end

  def recursive([], _map), do: false
  def recursive([h | tail], map) do
    if map[h] do
      true
    else
      recursive(tail, Map.put(map, h, true))
    end
  end
end

File: Arrays & Hashing/1. Contains Duplicate/solution.go
package main

import "fmt"

func main() {
	nums := []int{1, 2, 3}
	fmt.Print(containsDuplicate(nums))
}

func containsDuplicate(nums []int) bool {
	exist_map := make(map[int]bool)
	for i := 0; i < len(nums); i++ {
		if exist_map[nums[i]] {
			return true
		} else {
			exist_map[nums[i]] = true
		}
	}
	return false
}
